tools/cli/commands/docker.js

```diff
@@ -3,19 +3,133 @@
  */
 import { spawnSync } from 'child_process';
 import chalk from 'chalk';
+import { createWriteStream, existsSync, copyFileSync, readFileSync, writeFileSync } from 'fs';
+
+const dockerFolder = `tools/docker`;
 
 /**
- * Default handler for the monorepo Docker commands.
+ * Sets default options that are common for most of the commands
  *
- * @param {object} argv - Arguments passed.
+ * @param {Object} yargs Yargs
+ * @return {Object} Modified Yargs object
  */
-const defaultDockerCmdHandler = argv => {
-	const args = process.argv.slice( 4 ); // node pnpm run docker <cmd> [args..]
+const defaultOpts = yargs =>
+	yargs
+		.option( 'type', {
+			alias: 't',
+			describe: 'Container type',
+		} )
+		.option( 'name', {
+			alias: 'n',
+			describe: 'Project name',
+		} )
+		.option( 'port', {
+			alias: 'p',
+			describe: 'WP port',
+		} );
+
+/**
+ * Gets a project name from the passed aguments. Defaults to 'dev' if not specified.
+ *
+ * @param {Object} argv Yargs
+ * @return {string} Project name
+ */
+const getProjectName = argv => {
+	let project = 'dev';
+	if ( argv.type === 'e2e' ) {
+		project = argv.name ? argv.name : 'e2e';
+	}
+
+	return project;
+};
+
+/**
+ * Builds a map of ENV variables for specified configuration
+ *
+ * @param {Object} argv Yargs
+ * @return {Object} key-value pairs of ENV variables
+ */
+const buildEnv = argv => {
+	const envOpts = { PORT: 80 };
+	if ( argv.type === 'e2e' ) {
+		envOpts.PORT_WORDPRESS = argv.port ? argv.port : 8889;
+	}
+
+	envOpts.COMPOSE_PROJECT_NAME = getProjectName( argv );
+	return envOpts;
+};
+
+/**
+ * Creates an .env file
+ */
+const setEnv = () => {
+	createWriteStream( `${ dockerFolder }/.env`, {
+		flags: 'a',
+	} );
+};
+
+/**
+ * Creates a default extras file if needed.
+ */
+const setExtras = () => {
+	const extrasFile = `${ dockerFolder }/compose-extras.yml`;
+	const extrasSampleFile = `${ dockerFolder }/compose-extras.yml.sample`;
+
+	if ( ! existsSync( extrasFile ) ) {
+		copyFileSync( extrasSampleFile, extrasFile );
+	}
+};
+
+/**
+ * Generates Volumes compose file
+ *
+ * @param {Object} argv Yargs
+ */
+const setVolumes = argv => {
+	const volumesFile = `${ dockerFolder }/compose-volumes.yml`;
+	const volumesBuiltFile = `${ dockerFolder }/compose-volumes.built.yml`;
+	const sampleFile = `${ dockerFolder }/compose-volumes.yml.sample`;
+
+	if ( ! existsSync( volumesFile ) ) {
+		copyFileSync( sampleFile, volumesFile );
+	}
```

Comment:
You saying that we probably should re-create the file on every run?

TBH, I'd extract the configuration into a formal config file (json/yaml) based, and then parse it in cli. But I'm kinda on the fence here, since it will naturally lead us to building the `compose` file dynamically, which not be a good idea.

Reply:
No, don't recreate it on every run. The formal config file would be the way to go.

We already do recreate tools/docker/compose-volumes.built.yml on every run. The change would be to create it more smartly.